Tests: Event Store. Покрытие API модуля тестами

dr-wh0 opened the pull request:
# Описание изменений

Реализованы **интеграционные тесты API** модуля событий (_**Event Store**_) для всех эндпоинтов работы с событиями системы, включая получение списка событий и скачивание событий в формате CSV. В качестве инфраструктуры для тестов добавлены специализированные фикстуры для создания тестовых событий через RPC-клиент модуля. Тесты покрывают все основные сценарии работы, включая валидацию данных, проверку пагинации и фильтрацию, обработку edge cases.

# Ключевые изменения

**Тестовая инфраструктура:**
- Фикстуры в `conftest.py`:
  - `cleanup_events` - очищает все события до и после каждого теста
  - `created_event` - создает тестовое событие через RPC-клиент и возвращает его данные
  - `multiple_events` - создает несколько тестовых событий (3) для тестирования 

**Утилиты очистки:**
- `cleanup_all_events()` - удаляет все события из базы данных

**Реализованные тесты:**

*Получение событий:*
- `test_get_events_success` - успешное получение списка событий
- `test_get_events_empty` - получение событий из пустой БД (при отсутствии событий)
- `test_get_events_pagination` - тестирование пагинации
- `test_get_events_by_module_name` - фильтрация событий по модулю (в котором было создано событие)
- `test_get_events_by_different_module` - фильтрация по событий по другому модулю
- `test_get_events_by_module_with_multiple_events` - фильтрация при множестве событий
- `test_get_events_with_invalid_pagination` - обработка невалидных параметров пагинации
- `test_get_events_unauthorized` - проверка доступа без авторизации

*Скачивание событий в CSV:*
- `test_download_events_success` - успешное скачивание CSV-файла с событиями
- `test_download_events_empty` - скачивание из пустой БД (при отсутствии событий)
- `test_download_events_by_module_name` - скачивание с фильтрацией по модулю (в котором было создано событие)
- `test_download_events_by_different_module` - скачивание с фильтром по другому модулю
- `test_download_events_by_module_with_multiple_events` - скачивание нескольких событий
- `test_download_events_unauthorized` - проверка доступа без авторизации

**Общая структура файлов для тестов:**
```
└── tests/  
	├── conftest.py  # Фикстуры для тестов  
	└── api/  
		└── test_retrive.py  # GET /event/, /event/download
```

# Особенности

- Тесты используют RPC-клиент (`EventStoreServiceLayerRPCClient`) для создания тестовых событий, что обеспечивает приближенность к реальному использованию
- Полностью покрыты оба эндпоинта модуля Event Store
- Тестируются все основные сценарии, валидируется формат данных, пагинация, фильтрация по модулям
- Проверяется корректность формата CSV-файла (заголовки, структура, содержимое)
- Реализована очистка событий перед и после каждого теста

# Критерии приёмки

- [x] Сформирована структура файлов для тестов
- [x] Реализованы все тестовые сценарии для получения событий и скачивания CSV событий
- [x] Добавлены необходимые фикстуры для создания тестовых событий
- [x] Покрыты основные ошибки и edge cases
- [x] Добавлена [документация](https://github.com/Aerodisk/openvair-docs/pull/16) к тестам (в документации проекта)
- [x] Все тесты успешно проходят:
<img width="1360" height="568" alt="image" src="https://github.com/user-attachments/assets/7cc03c4f-f83a-40be-9811-c14a89fd203a" />


# Связанные задачи

- Issue #265 

### Также исправлен (и доработан) функционал, описанный в данных задачах:
- Issue #263
- Issue #264

После исправления эндпоинты модуля работают корректно.

dr-wh0 committed:
Add event_store API tests and required fixtures

dr-wh0 committed:
Refactor tests with ruff format

dr-wh0 committed:
Fix events download API endpoint

dr-wh0 committed:
Fix filter by module_name in event_store API

dr-wh0 committed:
Refactor test files names: test_retrive/test_get -> test_retrieve

VladimirTrue reviewed (commented):
В целом всё гуд :thumbsup:
Есть пара смущающих моментов, которые хотелось бы подправить, либо получить понять обоснование подхода

dr-wh0 committed:
Refactor tests after review:

- Add @validate_call decorator instead of switching to Dict in get_all_events_by_module()
- Remove ruff noqa ARG001
- Rename tests names (filtered)
- Add assert_csv_events_empty() helper function

dr-wh0 commented on openvair/modules/event_store/tests/api/test_retrieve.py:
```diff
@@ -0,0 +1,296 @@
+    assert data['items'][0]['object_id'] == created_event['object_id']
+
+
+def test_get_events_by_different_module(
```
> выполнив `assert created_event['module'] != module_name` мы чётко уверены, что `created_event`, создается не с тем же значением, которое мы даём для переменной `module_name`.

Может тогда достаточно чего-то вроде:

```python
def test_download_events_filtered_by_different_module(
    client: TestClient, created_event: Dict
) -> None:
    module_name = f'not_{created_event["module"]}'
    assert created_event['module'] != module_name  # Вот про такую проверку речь?
    response = client.get(f'/event/download?module_name={module_name}')
    assert response.status_code == status.HTTP_200_OK
    assert_csv_events_empty(response)
```
?

Просто если мы создаём ивент прямо в тесте, условно вот так:

```python
def test_download_events_filtered_by_different_module(
    client: TestClient, created_event: Dict
) -> None:
    module_name = 'test_module'
    event_data = {
        'module': module_name,
        'object_id': str(uuid4()),
        'user_id': str(uuid4()),
        'event': 'test_event',
        'information': 'test_info',
    }
    event_client = EventStoreServiceLayerRPCClient(event_data['module'])
    event_client.add_event(
        object_id=event_data['object_id'],
        user_id=event_data['user_id'],
        event=event_data['event'],
        information=event_data['information'],
    )
    different_module_name = f'not_{event_data["module"]}'
    assert event_data['module'] != different_module_name
    response = client.get(
        f'/event/download?module_name={different_module_name}'
    )
    assert response.status_code == status.HTTP_200_OK
    assert_csv_events_empty(response)
```

То появляется вопрос о том, зачем вообще нужна фикстура `created_event`, которая используется и в более общих тестах, например в `test_get_events_success` или `test_download_events_filtered_by_module_name`. Мне просто кажется, что так тесты будут слишком неоднородными (если конкретно для тестов `filtered_by_different_module` создавать ивент в теле теста). А если убирать фикстуру вовсе - это сильно усложнит код без причины.

VladimirTrue commented:
approve :+1:

dr-wh0 committed:
Add module_name assert in filtered_by_different_module tests
